### DISCUSSION TITLE
Hfss3dLayout -- How to edit Port Excitations in PyAEDT?

### DISCUSSION THREAD

**Discussion by j4mag** [ansys/pyaedt#5535] [Q&A] [answered]
This is a fairly trivial operation with the GUI: Excitations > "Port Excitations..." > Magnitude / Phase are available to edit for each port.

How can this be replicated using PyAEDT? It is required for generating the correct fields with `h3d_app.post.create_fieldplot_layers(...)`

**Comment by j4mag** [comment]
related, but for Hfss: https://github.com/ansys/pyaedt/discussions/3471

**Reply by Samuelopez-ansys** [reply, marked as answer]
Hi @j4mag ,

You can use the method edit_source_from_file if you need a frequency-dependent Magnitude and phase.

or you can directly set the value:

app.excitation_objects["Port1"].properties["Magnitude"] = "2V"
